version: '3.7'
services:
  # POSTGRES DATABASE
  db:  
    image: postgres:15
    environment:
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_DB=${POSTGRES_DB}
    restart: always             # run as a service
    volumes:
        - "${POSTGRES_DIR}:/var/lib/postgresql/data"
    networks:
      - odoo16net
  
  # ODOO SERVER
  odoo: 
    build: .
    platform: linux/amd64 
    ports:
      - "${ODOO_PORT}:8069"  # web access port
      - "8888:8888"   # debug port
    depends_on:
      - db
    tty: true
    command: python3 -m debugpy --listen 0.0.0.0:8888 /usr/bin/odoo --db_user=${POSTGRES_USER} --db_host=db --db_password=${POSTGRES_PASSWORD}
    volumes:
      - "${CUSTOM_ADDONS}:/mnt/custom-addons"
      - ./etc:/etc/odoo
    restart: always             # run as a service
    networks:
      - odoo16net
  
  # POSTGRES DATABASE ADMINISTRATION
  pgadmin: 
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      - PGADMIN_DEFAULT_EMAIL=${PGADMIN_DEFAULT_EMAIL}
      - PGADMIN_DEFAULT_PASSWORD=${PGADMIN_DEFAULT_PASSWORD}
    depends_on:
      - db
    ports:
      - "${PGADMIN_PORT}:80"
    networks:
      - odoo16net
    restart: always

networks:
  odoo16net:
    driver: bridge
